File: math/radix/converter.py


def convert(n :int, radix, capital=False) -> str :

		assert 1 <radix <37
		base = str()

		while(n):

			rem = n %radix
			base = (str(rem) if rem <10 else chr((55 if capital else 87) +rem)) + base
			n //= radix

		return base

File: math/radix/converter.c


char* uintConvert(uint n, int radix, bool capital)
{
	char* base = malloc(sizeof(char) *33);
	char* converted;
	int   i = 0;
	int   j = 0;
	int   rem;

	while(n)
	{
		rem = n %radix;
		*(base +i++) = (char)(rem + (rem <10 ? '0' : (capital ? 'A' : 'a') -10));
		n /= radix;
	}
	converted = malloc(sizeof(char) *(i-- +1));
	while(-1 <i) *(converted +j++) = *(base +i--);
	*(converted +j) = '\0';

	free(base);
	base = NULL;

	return converted;
}

